Examples/ObjectiveC/ShapeAnnotationsExample.m

```diff
@@ -0,0 +1,96 @@
+#import "ShapeAnnotationsExample.h"
+@import Mapbox;
+#import <malloc/malloc.h>
+
+NSString *const MBXExampleShapeAnnotations = @"ShapeAnnotationsExample";
+
+@interface ShapeAnnotationsExample () <MGLMapViewDelegate>
+
+@property CLLocationCoordinate2D pointCoordinate;
+@property (assign) CLLocationCoordinate2D *lineCoordinates;
+@property (assign) CLLocationCoordinate2D *polygonCoordinates;
+
+@end
+
+@implementation ShapeAnnotationsExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    MGLMapView *mapView = [[MGLMapView alloc] initWithFrame:self.view.frame];
+    mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    [mapView setCenterCoordinate:CLLocationCoordinate2DMake(45.520, -122.663) zoomLevel:13 animated:NO];
+    // Allow this ViewController class to be the reciever of `MGLMapViewDelegate` events.
+    mapView.delegate = self;
+    [self.view addSubview:mapView];
+
+    [self setupCoordinates];
+
+    // Three shapes annotations are created below: a point, line, and a polygon.
+    MGLPointAnnotation *pointAnnotation = [[MGLPointAnnotation alloc] init];
```

Comment:
It might be a bit neater to move the creation of these annotations to a separate method as well. You could probably make it work that way without needing to store the annotations as properties.